pkg/api/bzz.go

```diff
@@ -462,7 +469,12 @@ func (s *Service) downloadHandler(logger log.Logger, w http.ResponseWriter, r *h
 	if headers.Cache != nil {
 		cache = *headers.Cache
 	}
-	reader, l, err := joiner.New(r.Context(), s.storer.Download(cache), reference)
+
+	ctx := r.Context()
+	ctx = getter.SetStrategy(ctx, headers.Strategy)
+	ctx = getter.SetStrict(ctx, headers.FallbackMode)
+	ctx = getter.SetFetchTimeout(ctx, headers.ChunkRetrievalTimeout)
+	reader, l, err := joiner.New(ctx, s.storer.Download(cache), s.storer.Cache(), reference)
```

Comment:
storer.Cache() is not a reliable source for storing chunks for any operation, what is the purpose of using the cache here?

Reply:
what should I put here? there is no other putter available under `s.storer`

Reply:
what is the purpose of using the cache here?

Reply:
so cache must be used to store things locally that have no postage stamp. There are three ways to get there:
- evicted from reserve (either too distant, too cheap or expired)
- landed with us through retrieval and has no postage stamp (yet)
- and now also created with the help of parities,  no postage stamp available (yet).

Reply:
Lets get to the bottom of this. 

Reconstructed chunks have no postage stamp therefore they cannot be in the reserve. Putting them in a pinstore is wasteful and unclear in terms of expiry. 
The only possible place to put is the localstore cache.

The status of reconstructed chunks is similar to chunks obtained through retrieval which is also cached. 
The only thing we absolutely need to make sure of is that these cached chunks can and will be put in the reserve once they are offered by peers as part of pullsyncing as a chunk with a  valid postage stamp.

There is one caveat about using the cache is the intricate scenario described and resolved in this hackmdQq https://hackmd.io/@zelig/Bkqol64dp

Reply:
An even worse consideration is that the cache size is user-specified and can therefore be set to an arbitrarily small value, possibly ensuring that the reconstructed chunks have already been purged before they are needed again.

Reply:
I am introducing a recoder accordingly 
resolved in #4529